--- beid-2.6.0/admin/generic.py	2006-06-20 06:19:59.000000000 -0700
+++ beid-2.6.0/admin/generic.py.new	2008-12-27 19:10:04.000000000 -0800
@@ -536,7 +536,7 @@
 			source=src2
 
 		library_list = thisenv.SharedLibrary(target, source)
-		lafile_list  = thisenv.LaFile(target, library_list)
+#		lafile_list  = thisenv.LaFile(target, library_list)
 
 		## Install the libraries automatically
 		if not thisenv.has_key('NOAUTOINSTALL') and not noinst:
